payroll/models/hr_payslip.py

```diff
@@ -425,6 +433,67 @@ def get_inputs(self, contracts, date_from, date_to):
                 )
         return res
 
+    def get_payroll_dict(self, contracts):
+        """Setup miscellaneous dictionary values.
+        Other modules may overload this method to inject discreet values into
+        the salary rules. Such values will be available to the salary rule
+        under the `payroll.` prefix.
+
+        This method is evaluated once per payslip.
+        :param contracts: Recordset of all hr.contract records in this payslip
+        :return: a dictionary of discreet values and/or Browsable Objects
+        """
+        self.ensure_one()
+        return {}
+
+    def get_baselocaldict(self, contracts):
+        """Basic dictionary values that are useful in most salary rules. Inherited
+        classes that overload this method should use the name of the module as
+        the dictionary key.
+
+        This method is evaluated once per payslip.
+
+        :param contracts: Recordset of all hr.contract records in this payslip
+        :return: a dictionary of discreet values and/or Browsable Objects
+        """
+        self.ensure_one()
+
+        worked_days_dict = {}
+        inputs_dict = {}
+        payslip = self
+        for worked_days_line in payslip.worked_days_line_ids:
+            worked_days_dict[worked_days_line.code] = worked_days_line
+        for input_line in payslip.input_line_ids:
+            inputs_dict[input_line.code] = input_line
+        inputs = InputLine(payslip.employee_id.id, inputs_dict, self.env)
+        worked_days = WorkedDays(payslip.employee_id.id, worked_days_dict, self.env)
+        payslips = Payslips(payslip.employee_id.id, payslip, self.env)
+        payroll_dict = BrowsableObject(
+            payslip.employee_id.id, payslip.get_payroll_dict(contracts), self.env
+        )
+
+        baselocaldict = {
+            "payslip": payslips,
+            "worked_days": worked_days,
+            "inputs": inputs,
+            "payroll": payroll_dict,
+        }
+        return baselocaldict
+
+    def get_contract_dict(self, contract, contracts):
```

Comment:
Also add examples and more documentation here.

Reply:
Comments in the code